crates/iceberg-catalog/src/service/hooks.rs

```diff
@@ -0,0 +1,280 @@
+    ) {
+        for hook in &self.0 {
+            hook.drop_view(
+                warehouse_id,
+                parameters.clone(),
+                drop_params.clone(),
+                view_ident_uuid,
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+
+    async fn rename_view(
+        &self,
+        warehouse_id: WarehouseIdent,
+        prefix: Option<Prefix>,
+        view_ident_uuid: ViewIdentUuid,
+        request: Arc<RenameTableRequest>,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        for hook in &self.0 {
+            hook.rename_view(
+                warehouse_id,
+                prefix.clone(),
+                view_ident_uuid,
+                request.clone(),
+                request_metadata.clone(),
+            )
+            .await;
+        }
+    }
+}
+
+#[async_trait::async_trait]
+pub trait EndpointHooks: Send + Sync {
+    async fn commit_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _prefix: Option<Prefix>,
+        _request: Arc<CommitTransactionRequest>,
+        _table_ident_map: Arc<HashMap<TableIdent, TableIdentUuid>>,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+        // Default implementation does nothing
+    }
+
+    async fn drop_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _parameters: TableParameters,
+        _drop_params: DropParams,
+        _table_ident_uuid: TableIdentUuid,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+    async fn register_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _parameters: NamespaceParameters,
+        _request: Arc<RegisterTableRequest>,
+        _table_ident_uuid: TableIdentUuid,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+    async fn create_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _parameters: NamespaceParameters,
+        _table_ident_uuid: TableIdentUuid,
+        _request: Arc<CreateTableRequest>,
+        _data_access: DataAccess,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+
+    async fn rename_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _prefix: Option<Prefix>,
+        _table_ident_uuid: TableIdentUuid,
+        _request: Arc<RenameTableRequest>,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+
+    async fn create_view(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _view_ident_uuid: ViewIdentUuid,
+        _parameters: NamespaceParameters,
+        _request: Arc<CreateViewRequest>,
+        _data_access: DataAccess,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+        // Default implementation does nothing
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn commit_view(
+        &self,
+        _warehouse_id: WarehouseIdent,
+        _view_ident_uuid: ViewIdentUuid,
```

Comment:
Lets called all `xxx_ident_uuid` parameters simply `xxx_id`, i.e. `view_id`.
I already fixed project_id globally, but not yet the view / table ids. Then we just have:
`xxx_ident`: Iceberg identifier
`xxx_id`: Unique id